### PR TITLE
Fix function dateReadBack bug

### PR DESCRIPTION
**Pull request by Danieladu**
## Description
Function `dateReadBack` is broken. This PR fixes this issue.

### CODE REVIEW

File: libraries/adaptive-expressions/tests/expressionParser.test.js [hunk 1]
```diff
@@ -962,10 +961,6 @@ const generateParseTest = (input, expectedOutput, expectedRefs) => () => {
 }
 
 describe('expression parser functional test', () => {
-    describe("tt", () => {
```
Comment (joshgummersall) [inline libraries/adaptive-expressions/tests/expressionParser.test.js:965]: Why is this being removed?

File: libraries/adaptive-expressions/src/builtinFunctions/dateReadBack.ts [hunk 2]
```diff
@@ -36,9 +36,9 @@ export class DateReadBack extends ExpressionEvaluator {
             const dateFormat = 'YYYY-MM-DD';
             ({ value, error } = InternalFunctionUtils.parseTimestamp(args[0]));
             if (!error) {
-                const timestamp1: Date = new Date(value.format(dateFormat));
+                const timestamp1: Date = new Date(moment(value).format(dateFormat));
                 ({ value, error } = InternalFunctionUtils.parseTimestamp(args[1]));
-                const timestamp2: string = value.format(dateFormat);
+                const timestamp2: string = moment(value).format(dateFormat);
```
Comment (joshgummersall) [inline libraries/adaptive-expressions/src/builtinFunctions/dateReadBack.ts:41]: No need to include `: string` that will be properly inferred. Same with `: Date` above and `: TimexProperty` below.

**Review by joshgummersall — CHANGES_REQUESTED** [review]
A couple of quick things otherwise looks good.

File: libraries/adaptive-expressions/tests/expressionParser.test.js (hunk 1 above)
Comment (Danieladu) [reply libraries/adaptive-expressions/tests/expressionParser.test.js:965]: This test was introduced by mistake before and used as a local test.

**Review by joshgummersall — APPROVED** [review]

**Pull request merged**